apparmor.d/groups/browsers/firefox

```diff
@@ -172,15 +174,22 @@ profile firefox @{exec_path} flags=(attach_disconnected) {
   owner /tmp/user/@{uid}/@{name}/* rwk,
   owner /tmp/@{name}/ rw,
   owner /tmp/@{name}/* rwk,
-  owner /tmp/* rw,
   owner /tmp/firefox_*/ rw,
   owner /tmp/firefox_*/* rwk,
   owner /tmp/mozilla_*/ rw,
   owner /tmp/mozilla_*/* rw,
-  owner /tmp/MozillaBackgroundTask-*-removeDirectory/ rw,
-  owner /tmp/MozillaBackgroundTask-*-removeDirectory/** rwk,
-  owner /tmp/Mozillato-be-removed-cachePurge-* k,
-  owner /tmp/Temp-@{uuid}/ rw,
+  owner /tmp/MozillaBackgroundTask-???????????????-removeDirectory/{**,} rw,
+  owner /tmp/MozillaBackgroundTask-???????????????-removeDirectory/.parentlock k,
+  owner /tmp/Mozillato-be-removed-cachePurge-??????????????? rwk,
+  owner /tmp/Mozilla@{uuid}-cachePurge-??????????????? rwk,
+  owner /tmp/Mozilla\{@{uuid}\}-cachePurge-??????????????? rwk,
+  owner /tmp/Temp-@{uuid}/{**,} rw,
+  owner /tmp/mozilla-temp-@{int} rw,
+  owner /tmp/@{rand8}.txt w,
+  owner /tmp/tmp-???.xpi rw,
+  owner /tmp/.xfsm-ICE-@{rand6} rw,
+  owner /tmp/tmpaddon r,
+  owner /tmp/* w,  # file downloads (to anywhere)
```

Comment:
?

Reply:
On every file download, FF attempts to write a file in `/tmp` with arbitrary name, regardless of actual saving directory. Read access is not needed.